src/Client/HttpClientTestFactory.php

```diff
@@ -28,18 +28,10 @@
  */
 class HttpClientTestFactory
 {
-    /** @var array */
-    private $transactionLog = [];
+    private array $transactionLog = [];
+    private MockHandler $mockHandler;
+    private static ?HttpClientTestFactory $instance = null;
```

Comment:
```suggestion
    private static HttpClientTestFactory $instance;
```
wie in `InMemoryAccessTokenCache`: wenn du Zeile 46 auf `if (!isset(self::$instance)) {` änderst, muss die Property nicht mehr _nullable_ sein und du kannst dir die Zuweisung des Default-Wertes sparen

Reply:
Hier funktioniert der Vorschlag